redfish-core/lib/pcie.hpp

```diff
@@ -292,6 +300,87 @@ static inline void addLinkStatusToPcieDevice(
         std::array<const char*, 1>{pcieDeviceInterface});
 }
 
+/**
+ * @brief Add link status and health property to PCIe device
+ *
+ * @param[in, out]  asyncResp   Async HTTP response.
+ * @param[in]       pcieDevice  PCIe device name/ID.
+ */
+inline void addLinkStatusToPcieDevice(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& pcieDevice)
+{
+    auto linkStatus =
+        [asyncResp](const std::string& pcieSlotPath,
+                    const dbus::utility::MapperServiceMap& serviceMap) {
+            getPcieSlotLinkStatus(asyncResp, pcieSlotPath, serviceMap);
+        };
+
+    findPcieSlotPath(asyncResp, pcieDevice, std::move(linkStatus));
+}
+
+/**
+ * @brief Set linkReset property
+ *
+ * @param[in, out]  asyncResp       Async HTTP response.
+ * @param[in]       pcieSlotPath    PCIe slot path.
+ * @param[in]       serviceMap      A map to hold Service and corresponding
+ * interface list for the given cable id.
+ * @param[in]       linkReset       Flag to reset.
+ */
+inline void handleLinkReset(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
```

Comment:
Minor minor nitpick:

Do we want to add code here that checks if the resetLink value is the same as the previous resetLink value? If previous == current, we can skip dbus call?

Reply:
We still need a d-bus call to get resetLink right?